Stand 106/code enhancements

deenoBaji opened the pull request:
Here, I have enhanced the code by addressing issues with minimal (to be reviewed) or zero impact on existing workflows

trevor-james-nangosha commented:
@deenoBaji Could you please include the link to the issue in the PR description. And also if you could change the title to something like `STAND-106: Code Enhancements`

deenoBaji commented:
> @deenoBaji Could you please include the link to the issue in the PR description. And also if you could change the title to something like `STAND-106: Code Enhancements`
Thanks for the review. I changed the pull request name accordingly and added the issue to the description as well.
